<?xml version="1.0"?>
<launch>
    <arg name="gui" default="false"/>
    <arg name="sim" value="true"/>
    <arg name="reverse_omega" default="false"/>
    <arg name="use_spiderkiller_joy_stick" default="false"/>
    <include file="$(find docking_sim)/launch/link.launch">
        <arg name="gui" value="$(arg gui)"/>
    </include>

    <node pkg="rap_controller" type="rap_controller.py" name="rap_controller" output='screen'>
        <param name="robot_name" value="car1"/>
        <param name="sim" value="$(arg sim)"/>
        <param name="reverse_omega" value="$(arg reverse_omega)"/>
        <param name="ctl_frequency" value="30"/>
        <!-- Tf frame id  -->
        <param name="map_frame" value="map"/>
        <param name="map_peer_frame" value="map"/>
        <param name="big_car_frame" value="base_link"/>
        <param name="big_car_peer_frame" value="base_link"/>
        <param name="base_link_frame" value="car1"/>
        <param name="base_peer_frame" value="car2"/>
        <!-- Topic name -->
        <param name="cmd_vel_topic_leader" value="/cmd_vel_car1"/>
        <param name="cmd_vel_topic_follower" value="/cmd_vel_car2"/>
    </node>

    <node pkg="joy" type="joy_node" name="joy">
        <param name="autorepeat_rate" value="2.0"/>
        <param name="deadzone" value="0.05" />
    </node>

    <node pkg="rap_controller" type="teleop_joy.py" name="teleop_joy" output='screen'>
        <param name="vx_max" value="0.2"/>
        <param name="vy_max" value="0.2"/>
        <param name="wz_max" value="0.5"/>
        <param name="use_spiderkiller_joy_stick" value="$(arg use_spiderkiller_joy_stick)"/>
    </node>
</launch>
